components/DetailsDashboardList.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { useState } from 'react';
+import styles from './DetailsCSS.module.css';
+import { getStudentTotalChallengesCompletedInBlock } from '../util/api_proccesor';
+
+export default function DetailsDashboardList(props) {
+  const [hideDetails, setHideDetails] = useState(true);
+  const [buttonText, setButtonText] = useState('View details');
+
+  const handleShowDetails = () => {
+    if (hideDetails) {
+      setHideDetails(false);
+    } else {
+      setHideDetails(true);
+    }
```

Comment:
nit: I think we can reduce this to 
`
const handleShowDetails = () => {
setHideDetails(!hideDetails) } ...
` 
just to shorten the function a little bit